src/NServiceBus.RavenDB/Internal/TransactionRecoveryStorageVerifier.cs

```diff
@@ -0,0 +1,25 @@
+﻿namespace NServiceBus.RavenDB.Internal
+{
+    using Raven.Client;
+    using Raven.Client.Document;
+    using Raven.Client.Document.DTC;
+
+    static class TransactionRecoveryStorageVerifier
+    {
+        public static void ReplaceStorageIfNotSetByUser(IDocumentStore store)
+        {
+            var docStore = store as DocumentStore;
+
+            if (docStore == null)
+                return;
+
+            var currentStorage = docStore.TransactionRecoveryStorage;
+
+            // VolatileOnlyTransactionRecoveryStorage is the default. If set to anything else, don't change it!
+            if (currentStorage == null || currentStorage is VolatileOnlyTransactionRecoveryStorage)
```

Comment:
VolatileOnlyTransactionRecoveryStorage is actually the default for a DocumentStore. While it could (potentially, possibly?) be null, and that's included here for completeness against NRE, there's no way to tell if a user selected VolatileOnlyTransactionRecoveryStorage on purpose or is just accepting the default.

But either way, it's a dumb choice. It's the DTC transaction recovery equivalent of in-memory persistence.